[LimYanLing] iP

yanlings opened the pull request:
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Duke frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
-  ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from here.
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the main method:
```
public class Launcher{
    public static void main(String[] args) throws DukeException, IOException {
        Application.launch(Main.class,args);
    }
}
```

Commit:
Add Gradle support

yanlings committed:
Level-1

yanlings committed:
Level-2

yanlings committed:
Level-3

yanlings committed:
Level-4

yanlings committed:
Level-5

yanlings committed:
Level-6

yanlings committed:
A-TextUiTesting

yanlings committed:
A-TextUiTesting

yanlings committed:
Level 6

yanlings committed:
Level-6

yanlings committed:
Level-6

yanlings committed:
A-TextUiTesting

yanlings committed:
Level-6

yanlings committed:
A-MoreOOP

yanlings committed:
A-Packages

yanlings committed:
A-Package

yanlings committed:
Merge pull request #1 from nus-cs2103-AY2223S2/add-gradle-support

Add Gradle support

yanlings committed:
A-Gradle

yanlings committed:
A-JUnit

yanlings committed:
A-Jar

yanlings committed:
A-JavaDoc

yanlings committed:
A-CodingStandard

yanlings committed:
Level-9

Emrysil commented on src/main/java/duke/handleInvalidArgs.java:
```diff
@@ -0,0 +1,72 @@
+/**
+ * This is a class to handle the invalid number of arguments for each of the class
+ */
+public class handleInvalidArgs {
```
Perhaps can rename to InvalidArgsHandler?since we're suppose to use nouns for class names:)

Emrysil commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,74 @@
+package duke;
+
+import java.io.*;
```
Maybe try to specify the class being imported instead of just importing *

Emrysil commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,56 @@
+ * This is a class for Event action
+ */
+import duke.Task;
+public class Event extends Task {
```
I think import statements should be grouped together

Emrysil commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,56 @@
+     */
+    public Event(String description, String date)  {
+        super(description);
+        String[] fromandto = date.split("/");
```
Maybe rename to fromAndTo to following the camelCase naming convention

Emrysil commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,56 @@
+     */
+    @Override
+    public String toSaveString() {
+        String timeStr = this.start.format(DateTimeFormatter.ofPattern("HH:mm, MMM dd yyyy"));
```
Maybe rename to timeStart? timeStr can be a bit misleading as it tends to be interpreted as timeString

Emrysil commented on src/test/java/EventTest.java:
```diff
@@ -0,0 +1,15 @@
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
```
I might be wrong but I think you don't have to separate the two imports

yanlings committed:
branch-level-7

yanlings committed:
Level-10

yanlings committed:
Level-10

yanlings committed:
GUI does not return anything when user does mark, unmark or delete.

It needs to be change because the GUI would seem unresponsive if it did not return anything.

3 new methods have been added into UI.java
Add stringMark()
Add stringUnmark()
Add stringDelete()

By separating the methods, it improves readability of the code, and allows the GUI to respond accordingly.

yanlings committed:
A-Assertions

yanlings committed:
Merge pull request #3 from yanlings/branch-A-Assertions

A-Assertions

yanlings committed:
Update classes according to code quality

Shorten getResponse() method as it was too long
Create methods such as toPrintToDoString, toPrintEventString, toPrintDeadlineString

yanlings committed:
Merge pull request #4 from yanlings/branch-A-CodeQuality

Update classes according to code quality

yanlings committed:
Update classes according to C-Tagging

Created an addition condition in Duke.java to create tag task.
Created methods like stringTag and handleTag to add the string representation of the tag

yanlings committed:
Updated README.md

yanlings committed:
no message

yanlings committed:
Created Jar File for final release
